Pull a statically linked RITA binary from github

Zalgo2462 opened the pull request:
Changes installer.sh to pull a statically linked RITA binary for Linux x86-64 from github.com rather than installing go and building the rita binary.


Uses the github releases API to grab the latest release version.

Requires rita, config.yaml, and LICENSE to be present in every release here forward.

ethack commented on install.sh:
```diff
@@ -321,57 +219,50 @@ __install_mongodb() {
+	_RITA_CONFIG_URL="$_RITA_DOWNLOAD_URL/config.yaml"
+	_RITA_LICENSE_URL="$_RITA_DOWNLOAD_URL/LICENSE"
+	
+	curl -sSL $_RITA_BINARY_URL -o $_BIN_PATH/rita
```
The other two `curl` commands below quote the params. Might as well here for consistency right?

ethack commented:
The readme should be updated to reflect how install.sh is called.

I'll checkout the branch and start running through a test now.

ethack commented on install.sh:
```diff
@@ -321,57 +219,50 @@ __install_mongodb() {
 	if [ -f "$_CONFIG_PATH/config.yaml" ]; then
 		printf "$_SUBITEM Backing up your current RITA config: $_CONFIG_PATH/config.yaml -> $_CONFIG_PATH/config.yaml.old \n"
-		$_ELEVATE mv -f "$_CONFIG_PATH/config.yaml" "$_CONFIG_PATH/config.yaml.old"
+		mv -f "$_CONFIG_PATH/config.yaml" "$_CONFIG_PATH/config.yaml.old"
```
I just realized this is a little backwards from what I was thinking.

Thoughts on this instead?
* If `config.yaml` exists don't touch it.
* Instead download new config to `config.yaml.new` overwriting previous `config.yaml.new` if necessary

That will not wipe the current config or any customizations to it.  It will still work as long as the new config doesn't have breaking changes.

ethack replied:
@william-stearns Thoughts on the proposed change here?  Does that sound like a good solution?

william-stearns replied:
I agree with Ethan's "Thoughts on this instead?" steps.

Zalgo2462 commented:
I'd like to update the Readme Contributing and installation documentation as well

Zalgo2462 commented:
Updated the documentation, (see the docs folder, readme and contributing files), and changed the installer to pull config.yaml and LICENSE from the repo rather than the release page. 

Fixed #189 with Bill's help

Zalgo2462 commented:
Just found an issue with reinstallation hold on for one second...

Zalgo2462 commented:
LGTM
